### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
-Fix activation conditionals for Pwing thickness/HP/Mass tweak
﻿-Add whitelist option for guns/rockets/lasers to report hits in the competition marquee.
-Fix heatseeker missiles fired from reloadable rails occasionally losing lock on firing
-add toggle for setting maxAltitude and GM Kill limits to ASl or AGL.
-Fix issue causing Web API to not report scores.
-HP no longer rounds MM patched values.
-Steel Hull Material melting point increased.
-Weapon burst length/fire angle overrides now respect symmetry.
-Add Kill timer for GM Altitude to give violators a chance to get back to safe altitude instead of immediate kill.
-Add Infinite Propellant Gamemode toggle to BDA settings menu.
-Fix NRE spam when trying to fire Modular Missiles.
-Fix NRE when firing Modular Missiles and Missile Icons are enabled.
-Heatseekers now target the hottest part on a craft, not the CoM
-'Run Tournament' when using WayPoint Mode now properly spawns you at the start of the selected WP course

### CODE REVIEW

File: BDArmory/Control/BDModulePilotAI.cs [hunk 1]
```diff
@@ -1376,7 +1376,8 @@ public void OnAutoTuneChanged(BaseField field, object obj)
             pidAutoTuning.ResetMeasurements();
 
             SetAutoTuneFields();
-            MaintainFuelLevels(autoTune); // Prevent fuel drain while auto-tuning.
+            //MaintainFuelLevels(autoTune); // Prevent fuel drain while auto-tuning.
+            CheatOptions.InfinitePropellant = autoTune;
```
Comment (BrettRyland) [inline BDArmory/Control/BDModulePilotAI.cs:1380]:
```csharp
CheatOptions.InfinitePropellant = autoTune || BDArmorySettings.INFINITE_FUEL; // Prevent fuel drain while auto-tuning.
```

**Comment by BrettRyland** [comment]
If the above comments are all fine, I can make the changes myself and merge the PR.

File: BDArmory/Ammo/PooledBullet.cs [hunk 2]
```diff
@@ -1651,6 +1653,15 @@ public void UpdateWidth(Camera c, float resizeFactor)
 
         public void KillBullet()
         {
+            if (HEType == PooledBulletTypes.Slug && partsHit.Count > 0)
+            {
+                if (ProjectileUtils.isReportingWeapon(sourceWeapon) && BDACompetitionMode.Instance.competitionIsActive)
+                {
+                    string msg = $"{partsHit[0].vessel.GetName()} was nailed by {sourceVesselName}'s {sourceWeapon.partInfo.title} at {initialHitDistance:F3}, damaging {partsHit.Count} parts.";
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/Ammo/PooledBullet.cs:1660]: Ah, forgot to comment that out. Was testing to see which read better, ammo or weapon name.

File: BDArmory/UI/BDArmorySetup.cs [hunk 3]
```diff
@@ -985,7 +986,7 @@ void WindowBDAToolbar(int windowID)
             GUI.Label(new Rect(_windowMargin + _buttonSize, _windowMargin, columnWidth - 2 * _windowMargin - numberOfButtons * _buttonSize, _windowMargin + _buttonSize), StringUtils.Localize("#LOC_BDArmory_WMWindow_title") + "          ", kspTitleLabel);
 
             // Version.
-            GUI.Label(new Rect(columnWidth - _windowMargin - (numberOfButtons - 1) * _buttonSize - 100, 23, 57, 10), Version, waterMarkStyle);
+            GUI.Label(new Rect(columnWidth - _windowMargin - (numberOfButtons - 1) * _buttonSize - 1200, 23, 90, 10), Version + " SI", waterMarkStyle);
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/UI/BDArmorySetup.cs]: Yes. Looks like I missed removing it when I uploaded the latest pulls to git.

File: BDArmory/Targeting/TargetSignatureData.cs [hunk 4]
```diff
@@ -55,12 +56,13 @@ public TargetSignatureData(Vessel v, float _signalStrength)
             // vessel never been picked up on radar before: create new targetinfo record
             if (targetInfo == null)
             {
+                if (VesselModuleRegistry.GetMissileFire(v));
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/Targeting/TargetSignatureData.cs]: No.

**Comment by SuicidalInsanity** [comment]
Yes, comments are all fine.
(probably should have read to the end first before making changes... :P)

**Pull request merged**